### PR TITLE
test: mapped and reenabled pool edge cases

### PR DESCRIPTION
**Pull request by ivansolomonoff**
## Security Checklist

- [ ] 1. Re-Entrancy
- [ ] 2. Arithmetic Over/Under Flows
- [ ] 3. Unexpected Ether
- [ ] 4. Delegatecall
- [ ] 5. Default Visibilities
- [ ] 6. Entropy Illusion
- [ ] 7. External Contract Referencing
- [ ] 8. Short Address/Parameter Attack (off chain)
- [ ] 9. Unchecked CALL Return Values
- [ ] 10. Race Conditions / Front Running
- [ ] 11. Denial Of Service (DOS)
- [ ] 12. Block Timestamp Manipulation
- [ ] 13. Constructors with Care
- [ ] 14. Uninitialized Storage Pointers
- [ ] 15. Floating Points and Precision
- [ ] 16. Tx.Origin Authentication
- [ ] 17. Address.isContract Re-Entrancy via Constructor

### ⚠️ NOTES ⚠️

Make sure to think about each of these exploits in this PR.

### CODE REVIEW

File: test-suites/_base_reserve_configuration.spec.ts [hunk 1]
```diff
@@ -387,4 +411,386 @@ describe("ReserveConfiguration", async () => {
...
+      await pool
+        .connect(configSigner)
+        .initReserve(
+          config.xTokenAddress,
+          ZERO_ADDRESS,
+          config.variableDebtTokenAddress,
+          ZERO_ADDRESS,
+          ZERO_ADDRESS
+        )
+    );
```
Comment (Allen2898168) [inline test-suites/_base_reserve_configuration.spec.ts:475]: ` expect(x); ` inside It seems that no verification has been done

File: test-suites/_base_interest_rate_strategy.spec.ts [hunk 2]
```diff
@@ -238,6 +244,30 @@ describe("Interest Rate Tests", () => {
...
+        utils.parseEther("1")
+      );
+      await impersonateAccountsHardhat([configurator.address]);
+      const configSigner = await hre.ethers.getSigner(configurator.address);
+
+      expect(
+        await pool
+          .connect(configSigner)
+          .setReserveInterestRateStrategyAddress(dai.address, ZERO_ADDRESS)
+      );
```
Comment (Allen2898168) [inline test-suites/_base_interest_rate_strategy.spec.ts:266]:
```suggestion
      expect(
        await pool
          .connect(configSigner)
          .setReserveInterestRateStrategyAddress(dai.address, ZERO_ADDRESS)
      );
```
`expect(x);`  inside It seems that no verification has been done

File: test-suites/_base_reserve_configuration.spec.ts (hunk 1 above)
Comment (Allen2898168) [inline test-suites/_base_reserve_configuration.spec.ts:475]:
```suggestion
    expect(
      await pool
        .connect(configSigner)
        .initReserve(
          config.xTokenAddress,
          ZERO_ADDRESS,
          config.variableDebtTokenAddress,
          ZERO_ADDRESS,
          ZERO_ADDRESS
        )
    );
```
expect(x); inside It seems that no verification has been done

File: test-suites/_base_interest_rate_strategy.spec.ts (hunk 2 above)
Comment (Allen2898168) [inline test-suites/_base_interest_rate_strategy.spec.ts:266]:
```suggestion
      expect(
        await pool
          .connect(configSigner)
          .setReserveInterestRateStrategyAddress(dai.address, ZERO_ADDRESS)
      ).to.be.eq(true);
```
`expect(x);`  inside It seems that no verification has been done。  Can we add some verifications

File: test-suites/mint-to-treasury.spec.ts [hunk 3]
```diff
@@ -95,4 +95,20 @@ describe("Mint To Treasury", () => {
       "Invalid treasury balance after minting"
     );
   });
+
+  it("Call `mintToTreasury()` on a pool with an inactive reserve", async () => {
```
[the remark this answers is not in the document]
Comment (ivansolomonoff) [reply test-suites/mint-to-treasury.spec.ts:99]: yeah, that's pending to do in next PR, just moved that together with this to prepare for such migration.

File: test-suites/_base_interest_rate_strategy.spec.ts (hunk 2 above)
Comment (ivansolomonoff) [reply test-suites/_base_interest_rate_strategy.spec.ts:266]: well, that's the way Aave checked that simply a tnx succeeds, and maybe also kind of pointing the part under test, rather than leaving the test with no expect statement. As for `to.eq(true)`, I'm not convinced that's a better approach as would imply the function is returning a boolean, where is not. Maybe would be better, where possible, if the function emits some event, to use `.to.emit()`

File: test-suites/mint-to-treasury.spec.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ivansolomonoff) [reply test-suites/mint-to-treasury.spec.ts]: ok, did it within this same PR.

File: test-suites/_base_interest_rate_strategy.spec.ts (hunk 2 above)
Comment (Allen2898168) [reply test-suites/_base_interest_rate_strategy.spec.ts:268]: ok, As for to.eq(true) .Just an example

**Review by alannotnerd — APPROVED** [review]
